app/models/manageiq/providers/openstack/provider.rb

```diff
@@ -1,11 +1,11 @@
-class ProviderOpenstack < Provider
+class ManageIQ::Providers::Openstack::Provider < ::Provider
```

Comment:
:+1: Foreman already had a ..::Provider anyway.

As for the reasoning behind it... I've specifically kept `ManageIQ::Providers::Openstack` as a module, so that it can be freely used as a nesting parent across many files, without them all needing to know about the correct inheritance.

I believe it would also exacerbate the load cascade we currently have -- where, as soon as one model is loaded, a substantial number of related models all get pulled in at the same time, because of constant aliasing and/or our need for accurate `subclasses`.

I don't particularly mind this naming style ([which is not unique](http://api.rubyonrails.org/classes/ActiveRecord/ConnectionAdapters/PostgreSQLAdapter.html))... but otherwise, I think we'd need to address it by introducing a new noun; right now, "provider" is doing double-duty between this specific class and the whole potential-gem-full of classes that know about Openstack.